### PR TITLE
Fix byte offset mismatch in OpaqueData field in GET_CSR request message format 

### PR DESCRIPTION
**Pull request by Xiaohanjlll**
Fix: #2039

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_get_csr.c [hunk 1]
```diff
@@ -93,16 +93,16 @@ static libspdm_return_t libspdm_try_get_csr(libspdm_context_t *spdm_context,
     spdm_request->opaque_data_length = opaque_data_length;
     spdm_request->requester_info_length = requester_info_length;
 
-    if (opaque_data_length != 0) {
+    if (requester_info_length != 0) {
         libspdm_copy_mem(spdm_request + 1,
                          spdm_request_size - sizeof(spdm_get_csr_request_t),
                          (uint8_t *)opaque_data, opaque_data_length);
```
Comment (jyao1) [inline library/spdm_requester_lib/libspdm_req_get_csr.c:99]: You did not change opaque_data here.

How that can pass the unit test?

Comment (Xiaohanjlll) [reply library/spdm_requester_lib/libspdm_req_get_csr.c:99]: get_csr currently has no unit test related to this location, I will add them

File: unit_test/test_spdm_requester/get_csr.c [hunk 2]
```diff
@@ -42,7 +63,38 @@ libspdm_return_t libspdm_requester_get_csr_test_send_message(
...
+        /* Obtain the real spdm_request */
+        spdm_request =
+            (const spdm_get_csr_request_t *)((const uint8_t *)request +
+                                             sizeof(libspdm_test_message_header_t));
+
+        requester_info_length = spdm_request->requester_info_length;
+        opaque_data_length = spdm_request->opaque_data_length;
+
+        requester_info = (void*)((size_t)(spdm_request + 1));
+        result = libspdm_consttime_is_mem_equal(requester_info, right_req_info,
```
Comment (steven-bellock) [inline unit_test/test_spdm_requester/get_csr.c:83]: Can't you use [assert_memory_equal](https://api.cmocka.org/group__cmocka__asserts.html#gac933d8b75ebd55e88ca85963721c05d9)?

File: library/spdm_responder_lib/libspdm_rsp_csr.c [hunk 3]
```diff
@@ -113,9 +113,7 @@ libspdm_return_t libspdm_get_response_csr(libspdm_context_t *spdm_context,
                                                response_size, response);
     }
 
-    opaque_data = (void*)((size_t)(spdm_request + 1));
-
-    requester_info = (void*)(opaque_data + opaque_data_length);
+    requester_info = (void*)((size_t)(spdm_request + 1));
```
Comment (steven-bellock) [inline library/spdm_responder_lib/libspdm_rsp_csr.c:116]: Why the cast to `size_t`?

File: library/spdm_responder_lib/libspdm_rsp_csr.c [hunk 4]
```diff
@@ -141,6 +139,13 @@ libspdm_return_t libspdm_get_response_csr(libspdm_context_t *spdm_context,
             response_size, response);
     }
 
+    opaque_data = (void*)(requester_info + requester_info_length);
+    if(opaque_data == NULL) {
```
Comment (steven-bellock) [inline library/spdm_responder_lib/libspdm_rsp_csr.c:143]: How can `opaque_data` be `NULL`?

File: unit_test/test_spdm_requester/get_csr.c (hunk 2 above)
Comment (Xiaohanjlll) [reply unit_test/test_spdm_requester/get_csr.c]: Thanks.I have changed `libspdm_consttime_is_mem_equalto` to `assert_memory_equal`

File: library/spdm_responder_lib/libspdm_rsp_csr.c (hunk 3 above)
Comment (Xiaohanjlll) [reply library/spdm_responder_lib/libspdm_rsp_csr.c:115]: `const spdm_get_csr_request_t *spdm_request;`
It needs to be converted to a number first, and then to a pointer

File: library/spdm_responder_lib/libspdm_rsp_csr.c (hunk 4 above)
Comment (Xiaohanjlll) [reply library/spdm_responder_lib/libspdm_rsp_csr.c]: Thanks. 
`opaque_data `will not be `NULL`,  I have deleted it

File: library/spdm_responder_lib/libspdm_rsp_csr.c (hunk 3 above)
Comment (steven-bellock) [reply library/spdm_responder_lib/libspdm_rsp_csr.c:115]: `(spdm_request + 1)` is already a pointer. The `(size_t)` is not needed.

Comment (Xiaohanjlll) [reply library/spdm_responder_lib/libspdm_rsp_csr.c:115]: requester_info = (uint8_t *)(spdm_request + 1);
error: cast discards ‘const’ qualifier from pointer target type

Comment (steven-bellock) [reply library/spdm_responder_lib/libspdm_rsp_csr.c:115]: What compiler are you using?

**Comment by jyao1** [comment]
please resolve conflict.

File: library/spdm_responder_lib/libspdm_rsp_csr.c (hunk 3 above)
Comment (Xiaohanjlll) [reply library/spdm_responder_lib/libspdm_rsp_csr.c]: GCC compiler
I tried using CI to test more compilers，GCC reports error
https://github.com/Xiaohanjlll/libspdm/actions/runs/5052621802

**Pull request merged**